packages/nimble/inst/CppCode/tensorflow.cpp

```diff
@@ -178,3 +179,138 @@ void NimTf_Runner::NimTf_getOutput(NimArrBase<double>& nimArr) {
   TF_DeleteTensor(tensor);
   output_pos_ += 1;
 }
+
+void NimTf_Runner::NimTf_setInput(const std::vector<int>& dims,
+                                  const double* data) {
+  NIMERROR("TODO")
+}
+
+void NimTf_Runner::NimTf_runGradients() { NIMERROR("TODO") }
```

Comment:
Should this be `NimTf_Runner::NimTf_runGradient()` ?

Reply:
Good catch! (It's so nice to have someone else's eyes on the code :-)